dataflow/run_template/main.py

```diff
@@ -0,0 +1,121 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import google.auth
+import json
+from datetime import datetime
+
+"""Script to run a Dataflow template."""
+
+
+def run(template, parameters=None, project=None, job_name=None):
+    """Runs a Dataflow template."""
+
+    if project is None:
+        _, project = google.auth.default()
+
+    if job_name is None:
+        template_name = template.rsplit('/', 1)[-1]
+        timestamp = datetime.now().strftime('%Y%m%d-%H%M%S')
+        job_name = '{}-{}'.format(template_name, timestamp)
+
+    # [START run_template]
+    from googleapiclient.discovery import build
+
+    # template = 'gs://dataflow-templates/latest/Word_Count'
+    # parameters = {
+    #     'inputFile': 'gs://dataflow-samples/shakespeare/kinglear.txt',
+    #     'output': 'gs://{}/wordcount/outputs'.format(bucket),
+    # }
+    # project = 'your-gcp-project'
+    # job_name = 'unique-dataflow-job-name'
+
+    service = build('dataflow', 'v1b3')
+    request = service.projects().templates().launch(
+        projectId=project,
+        gcsPath=template,
+        body={
+            'jobName': job_name,
+            'parameters': parameters or {},
+        },
+    )
+    response = request.execute()
+    # [END run_template]
+    return response
+
+
+def run_template(request):
+    """HTTP Cloud Function.
+    Args:
+        request (flask.Request): The request object.
+        <http://flask.pocoo.org/docs/1.0/api/#flask.Request>
+    Returns:
+        The response text, or any set of values that can be turned into a
+        Response object using `make_response`
+        <http://flask.pocoo.org/docs/1.0/api/#flask.Flask.make_response>.
+    """
+    parameters = request.get_json(silent=True) or {}
+    parameters.update(request.args)
+    parameters.update(request.form)
+
+    template = parameters.pop('template')
+    project = parameters.pop('project', None)
+    job_name = parameters.pop('job_name', None)
+    response = run(
+        template=template,
+        parameters=parameters,
+        project=project,
+        job_name=job_name,
+    )
+    return json.dumps(response, separators=(':', ','))
+
+
+if __name__ == '__main__':
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('template', help='Dataflow template GCS path.')
+    parser.add_argument('--project', help='Google Cloud project ID.')
+    parser.add_argument('--job_name', help='Name for the Dataflow job.')
+    args, argv = parser.parse_known_args()
+
+    # Parse the arguments as a dictionary of type {str: str}.
+    #   Example:  --optionA A1 A2 --optionB B1 --optionA A3
+    #   Produces: {'optionA': 'A1 A2 A3', 'optionB': 'B1'}
+    parameters = {}
+    option = None
+    for arg in argv:
```

Comment:
This seems complex, is it possible to simplify it by making the command-line tool not support every possible parameter? At the least, argparse needs some custom help text to explain to the user that they can append arbitrary params, and a link to API documentation listing them. But we shouldn't hesitate to restrict the scope and capability of the samples, and link to that documentation in comments only.